source/fundamentals/typescript.txt

```diff
@@ -79,81 +80,51 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
 class in the
 :ref:`Find Multiple Documents Usage Example <node-driver-find-usage-example-code-snippet>`.
 
+.. _node-driver-limitations:
+
 Limitations
 -----------
 
-.. _node-driver-typescript-limitations-dot-notation:
+.. _node-driver-recursive-type:
 
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
+You cannot specify a recursive type as a type parameter for {+driver-long+}
+classes. Click the following tabs to see a recursive type and a code snippet that
+uses the recursive type and raises an error: 
 
 .. tabs::
 
-  .. tab:: Dot Notation
-     :tabid: dot-notation
-
-     The following code snippet does not raise a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
-
-     This is the error:
+   .. tab:: Recursive Type
+      :tabid: recursive-type
 
-     .. code-block:: text
+      .. code-block:: typescript
 
-        Type 'string' is not assignable to type 'number'.
+         interface RecursiveType {
+            r: RecursiveType | BaseCase;
+            i: number;
+         }
 
-Despite the lack of type safety, we still recommend that you use dot notation to
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
+         type BaseCase = "done"; 
 
-.. note:: Reason To Use Dot Notation
+   .. tab:: Code Snippet
+      :tabid: code-snippet
 
-   In the MongoDB Query Language, you must match a subdocument exactly
-   when specifying subdocuments in a query. Dot notation allows you to query
-   nested fields without matching subdocuments exactly.
+      The following code snippet uses a recursive type:
 
-   To show this behavior, lets say you have a collection containing
-   only the following document:
+      .. code-block:: typescript
 
-   .. code-block:: json
+         const database = client.db("test-db");
+         const col = database.collection<RecursiveType>("test-collection");
```

Comment:
Should there be some consistency in the code examples within the page? This placeholder is "test-collection", but the example in Extend Document is `"<your collection>"`.